src/components/display/fewlines/ConfirmationBox/ClickAwayListener.tsx

```diff
@@ -0,0 +1,10 @@
+import styled from "styled-components";
+
+export const ClickAwayListener = styled.div`
+  position: absolute;
+  height: 100%;
+  z-index: 1;
+  width: 100%;
+  left: 0;
+  top: 0;
+`;
```

Comment:
@Fenntasy What do you think of this (i.e. a `styled-component` with a `onClick` call)? 

Would you be ok with custom hook like this?

```ts
import React from 'react';

/**
 * Run the specified function whenever the user clicks somewhere outside of the
 * surface area of the supplied reference.
 *
 * Example: Closing a modal when the user clicks away from it.
 * @param ref RefObject of the element we want to react to outside clicks for.
 * @param handler The function to be executed when an outside click is registered.
 */
const useOnClickOutside = (
	ref: React.RefObject<any>,
	handler: (e: React.MouseEvent) => void
): void => {
	React.useEffect(() => {
		const listener = (event: React.MouseEvent) => {
			if (!ref.current || ref.current.contains(event.target)) {
				return;
			}

			handler(event);
		};

		document.addEventListener('mousedown', listener as any);

		return () => {
			document.removeEventListener('mousedown', listener as any);
		};
	}, [ref, handler]);
};

export default useOnClickOutside;
```

Reply:
I'm not against the principle but I'm always afraid of this kind of stuff: 
- do we support enough browser with it? 
- what happens if we click outside a modal but on a link, do we follow the link because the event has bubbled down?

There's always some edge cases that are annoying (with this particular stuff) that are a pain to debug.
That's why I always rely on a dom element that I create behind my modal, because I have more control over it.

But I insist, it's this particular case that makes me affraid, not this kind of custom hook

Reply:
@Hamza-A So after some discussion, we decided to keep your version.

Also
```suggestion
import styled from "styled-components";

export const ClickAwayListener = styled.div`
  position: fixed;
  width: 100%;
  height: 100vh;
  z-index: 1;
`;
```

Reply:
OK I kept `left: 0;` and `top: 0;` though, Otherwise, the listener would not cover the whole screen as illustrated by the red part 
![image](https://user-images.githubusercontent.com/33835824/95193556-4363b080-07d4-11eb-9ea9-23fd79d0adec.png)
a5a85c9

Reply:
> OK I kept left: 0; and top: 0; though

Ah yes, my bad.